File: jlpt/native_ios/Sources/Data/DTOs/AuthDTOs.swift
import Foundation

struct LoginRequestDTO: Encodable {
    let email: String
    let password: String
}

struct RegisterRequestDTO: Encodable {
    let email: String
    let password: String
    let displayName: String
    let targetLevel: JlptLevel?
}

struct RefreshRequestDTO: Encodable {
    let refreshToken: String
}

struct AuthResponseDTO: Decodable {
    let accessToken: String
    let refreshToken: String
    let userId: Int64
    let email: String
    let displayName: String
    let role: String
}

File: jlpt/native_android/app/src/main/java/com/jlpt/android/data/remote/dto/AuthDtos.kt
package com.jlpt.android.data.remote.dto

import com.jlpt.android.domain.model.JlptLevel
import kotlinx.serialization.Serializable

@Serializable
data class LoginRequestDto(
    val email: String,
    val password: String
)

@Serializable
data class RegisterRequestDto(
    val email: String,
    val password: String,
    val displayName: String,
    val targetLevel: JlptLevel? = null
)

@Serializable
data class RefreshRequestDto(
    val refreshToken: String
)

@Serializable
data class AuthResponseDto(
    val accessToken: String,
    val refreshToken: String,
    val userId: Long,
    val email: String,
    val displayName: String,
    val role: String
)
